pkg/cluster/internal/providers/podman/images.go

```diff
@@ -83,10 +83,25 @@ func pull(logger log.Logger, image string, retries int) error {
 
 // sanitizeImage is a helper to return human readable image name and
 // the podman pullable image name from the provided image
-func sanitizeImage(image string) (string, string) {
+func sanitizeImage(image string) (friendlyImageName, pullImageName string) {
+	defaulDomain := "docker.io/"
+	var remainder string
+
 	if strings.Contains(image, "@sha256:") {
 		splits := strings.Split(image, "@sha256:")
-		return splits[0], strings.Split(splits[0], ":")[0] + "@sha256:" + splits[1]
+		friendlyImageName = splits[0]
+		remainder = strings.Split(splits[0], ":")[0] + "@sha256:" + splits[1]
+	} else {
+		friendlyImageName = image
+		remainder = image
+	}
+
+	s := strings.Count(remainder, "/")
```

Comment:
I think instead we want to look for a period before the tag.
Even then that may not be safe actually, due to custom hostnames. Hmm.

Reply:
Addressed.